ext/phar/tests/phar_oo_008.phpt

```diff
@@ -31,45 +31,47 @@ foreach($f as $k => $v)
 
 class MyCSVFile extends SplFileObject
 {
-    function current()
+    function current(): string
```

Comment:
I don't get why this change is necessary. And `: array` return type should've been fine here, no?

Reply:
I did this because the return type of `SplFileObject::current()` is declared as `string|false`.

Reply:
Huh, I missed that change. It should be `string|array|false` as before -- why did you change that one?

Also wondering why it didn't result in an assertion failure due to type mismatch.

Reply:
I believe it was because I misunderstood your comment here: https://github.com/php/php-src/pull/7115#discussion_r664401884 So I thought that `current()` should also be changed along with `currentLine()`.

Reply:
> Also wondering why it didn't result in an assertion failure due to type mismatch.

Looks like we don't check this for calls via zend_call_function(). Working on it...

> I believe it was because I misunderstood your comment here: [#7115 (comment)](https://github.com/php/php-src/pull/7115#discussion_r664401884) So I thought that `current()` should also be changed along with `currentLine()`.

Nah, current() does need to return array -- that's the whole point of the READ_CSV mode.

Reply:
> Nah, current() does need to return array -- that's the whole point of the READ_CSV mode.

OK, thanks! As you can see, I'm not very familiar with SPL, but we all know why :D

Reply:
Check implemented in https://github.com/php/php-src/commit/ee65e92070f86f803a1a7b3ab9594dfe36c3b361.